spec/patron_spec.rb

```diff
@@ -0,0 +1,67 @@
+require 'rails_helper'
+
+RSpec.describe ::Patreon::Patron do
+
+  Fabricator(:oauth2_user_info) do
+    provider "patreon"
+    user
+  end
+
+  Fabricator(:user_custom_field) do
+    name "patreon_id"
+    user
+  end
+
+  let(:patrons) { { "111111" => { "email" => "foo@bar.com" }, "111112" => { "email" => "boo@far.com" },  "111113" => { "email" => "roo@aar.com" } } }
+  let(:pledges) { { "111111" => "100", "111112" => "500" } }
+  let(:rewards) { { "0" => { title: "All Patrons", amount_cents: "0" }, "4589" => { title: "Sponsers", amount_cents: "1000" } } }
+  let(:reward_users) { { "0" => ["111111", "111112"], "4589" => ["111112"] } }
+  let(:titles) { { "111111" => "All Patrons", "111112" => "All Patrons, Sponsers" } }
+
+  before do
+    ::Patreon.set("users", patrons)
+    ::Patreon.set("pledges", pledges)
+    ::Patreon.set("rewards", rewards)
+    ::Patreon.set("reward-users", reward_users)
+  end
+
+  it "should find rewarded user ids by rewards" do
+    ids = described_class.get_ids_by_rewards(rewards.keys)

```

Comment:
Hmm this is a little strange wouldn't the test blow up since this is a private method?

Reply:
I also expected a blow. But it worked. Maybe RSpec did the magic. If it is a inappropriate way I can move the method to public.

Reply:
Even so, private methods shouldn't need to be tested.

Reply:
It's a class method, so `private` has no effect. You'd need to use `private_class_method` to make it private...